### PR TITLE
[PropertyAccess] Remove deprecated code

### PR DESCRIPTION
**Pull request by malteschlueter**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

This remove deprecated code from PropertyAccess.

### CODE REVIEW

File: src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php [hunk 1]
```diff
@@ -123,19 +123,6 @@ public function testGetValueReturnsNullIfPropertyNotFoundAndExceptionIsDisabled(
         $this->assertNull($this->propertyAccessor->getValue($objectOrArray, $path), $path);
     }
 
-    /**
-     * @group legacy
-     * @dataProvider getPathsWithMissingProperty
-     */
-    public function testGetValueReturnsNullIfPropertyNotFoundAndExceptionIsDisabledUsingBooleanArgument($objectOrArray, $path)
```
Comment (jschaedl) [inline src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php:130]: Please also remove `ExpectDeprecationTrait` from this test:

https://github.com/symfony/symfony/blob/2c7ba63fe324bbe4206fb92f0c75b238ad4d52ce/src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php#L15

see https://github.com/symfony/symfony/pull/41298#discussion_r635621091

File: src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php [hunk 2]
```diff
@@ -123,19 +123,6 @@ public function testGetValueReturnsNullIfPropertyNotFoundAndExceptionIsDisabled(
...
-    {
-        $this->expectDeprecation('Since symfony/property-access 5.3: Passing a boolean as the fourth argument to "Symfony\Component\PropertyAccess\PropertyAccessor::__construct()" is deprecated. Pass a combination of bitwise flags as the second argument instead (i.e an integer).');
```
Comment (jschaedl) [inline src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php:132]: Please also remove `ExpectDeprecationTrait` from this test:

https://github.com/symfony/symfony/blob/2c7ba63fe324bbe4206fb92f0c75b238ad4d52ce/src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php#L15

see https://github.com/symfony/symfony/pull/41298#discussion_r635621091

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php [hunk 3]
```diff
@@ -86,53 +86,14 @@ class PropertyAccessor implements PropertyAccessorInterface
...
+     *                          or self::DISALLOW_MAGIC_METHODS for none
+     * @param int $throw        A bitwise combination of the THROW_* constants
+     *                          to specify when exceptions should be thrown
      */
-    public function __construct(/*int */$magicMethods = self::MAGIC_GET | self::MAGIC_SET, /*int */$throw = self::THROW_ON_INVALID_PROPERTY_PATH, CacheItemPoolInterface $cacheItemPool = null, /*PropertyReadInfoExtractorInterface */$readInfoExtractor = null, /*PropertyWriteInfoExtractorInterface */$writeInfoExtractor = null)
+    public function __construct(int $magicMethods = self::MAGIC_GET | self::MAGIC_SET, int $throw = self::THROW_ON_INVALID_PROPERTY_PATH, CacheItemPoolInterface $cacheItemPool = null, PropertyReadInfoExtractorInterface $readInfoExtractor = null, PropertyWriteInfoExtractorInterface $writeInfoExtractor = null)
     {
-        if (\is_bool($magicMethods)) {
-            trigger_deprecation('symfony/property-access', '5.2', 'Passing a boolean as the first argument to "%s()" is deprecated. Pass a combination of bitwise flags instead (i.e an integer).', __METHOD__);
```
Comment (jschaedl) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php:100]: You can remove `symfony/deprecation-contracts` from the composer.json file. See https://github.com/symfony/symfony/pull/41298#discussion_r635621710

**Comment by jschaedl** [comment]
Shall we update the component's CHANGELOG file accordingly?

**Comment by nicolas-grekas** [comment]
> Shall we update the component's CHANGELOG file accordingly?

yes please, please also rebase and take review comments into account of course

**Comment by nicolas-grekas** [comment]
Please rebase to restart tests :pray:

**Comment by nicolas-grekas** [comment]
Thank you @malteschlueter.

**Pull request merged**